test-data/unit/deps-statements.test

```diff
@@ -549,3 +549,101 @@ def f() -> int:
 <a> -> m
 <sys.platform> -> m.f
 <sys> -> m, m.f
+
+[case testOverload]
+from typing import overload
+
+class A: pass
+class B(A): pass
+
+@overload
+def f(x: B) -> B: ...
+@overload
+def f(x: A) -> A: ...
+
+def f(x: A) -> A:
+    g()
+    return x
+
+def g() -> None: pass
+[builtins fixtures/isinstancelist.pyi]
+[out]
+<m.A.__init__> -> <m.B.__init__>
+<m.A> -> <m.f>, m, m.A, m.B, m.f
```

Comment:
Remind me, why does A trigger m (the whole module?) but not B?

Reply:
It looks like it's because `A` is used as a base class in `B`, whereas `B` is not used as a base class. Base classes are considered to be in the top-level module scope. (Effectively both will trigger the module top level since `m.A` and `m.B` also include `m`.)